src/libraries/Microsoft.Extensions.Primitives/src/StringSegment.cs

```diff
@@ -671,6 +674,20 @@ private static void CheckStringComparison(StringComparison comparisonType)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool SliceArgumentsAreValid(int sourceLength, int start, int length)
+        {
+            // Copy of the internal MemoryExtensions.SliceArgumentsAreValid helper.
+            if (IntPtr.Size == 8)
```

Comment:
nit:

```suggestion
            if (nuint.Size == 8)
```

Reply:
It's notably predominant because we didn't have `nint`/`nuint` for the first 15 years

The repo in general prefers the use of the language keywords over BCL types (see 11 https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md).

Ideally we'd do a general code cleanup pass to normalize all of those

Reply:
IntPtrs typically represent unmanaged pointers in older code. Normalizing IntPtrs that represent unmanaged pointers to uints would not make the code more readable.

Reply:
> Normalizing IntPtrs that represent unmanaged pointers to uints would not make the code more readable.

I'd strongly disagree.

`IntPtr` isn't a pointer and `IntPtr.Size` vs `nint.Size` doesn't convey any different information. Arguably the use of `IntPtr.Size` is less readable and more confusing, especially when the subsequent code actually ends up using `nint`/`nuint` for the casts and arithmetic.

If we did want readability here, we should consider using `Environment.Is64BitProcess` instead, which correctly conveys the reason we've got 2 different code paths.